include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h

```diff
@@ -42,7 +43,7 @@ oneapi::dpl::__internal::__enable_if_host_execution_policy<typename ::std::decay
 exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 last1, InputIterator2 first2,
                                OutputIterator result, T init, BinaryPredicate binary_pred, Operator binary_op)
 {
-    const auto n = ::std::distance(first1, last1);
+    const auto n = last1 - first1;
```

Comment:
why ``std::distance`` is "bad"?

Reply:
I think we should not cheat ourself with `distance`, because `exclusive_scan_by_segment_impl` accepts juts random access iterators. And of course it is better (and more transparency) to get compile time error on the first line of the implementation than somewhere in "the code details" below....

Reply:
If the motivation not to use `std::distance` is to use this to make a better compiler error when `InputIterator1` is not a random access iterator, should we instead require random access iterators in the function signature with an `__enable_if_host_execution_policy_conditional`?

Reply:
Is the consensus to use `std::distance` with a static assert to ensure that only random access iterators are used?

Reply:
I think because there are conflicting opinions on this issue it is just best to be consistent with the current code which uses `::std::distance`.